lib/ttfunk/table/os2.rb

```diff
@@ -40,36 +41,245 @@ class OS2 < Table
+        (0x1200..0x137F) => 75,    (0x1380..0x139F) => 75,
+        (0x2D80..0x2DDF) => 75,    (0x13A0..0x13FF) => 76,
+        (0x1400..0x167F) => 77,    (0x1680..0x169F) => 78,
+        (0x16A0..0x16FF) => 79,    (0x1780..0x17FF) => 80,
+        (0x19E0..0x19FF) => 80,    (0x1800..0x18AF) => 81,
+        (0x2800..0x28FF) => 82,    (0xA000..0xA48F) => 83,
+        (0xA490..0xA4CF) => 83,    (0x1700..0x171F) => 84,
+        (0x1720..0x173F) => 84,    (0x1740..0x175F) => 84,
+        (0x1760..0x177F) => 84,    (0x10300..0x1032F) => 85,
+        (0x10330..0x1034F) => 86,  (0x10400..0x1044F) => 87,
+        (0x1D000..0x1D0FF) => 88,  (0x1D100..0x1D1FF) => 88,
+        (0x1D200..0x1D24F) => 88,  (0x1D400..0x1D7FF) => 89,
+        (0xF0000..0xFFFFD) => 90,  (0x100000..0x10FFFD) => 90,
+        (0xFE00..0xFE0F) => 91,    (0xE0100..0xE01EF) => 91,
+        (0xE0000..0xE007F) => 92,  (0x1900..0x194F) => 93,
+        (0x1950..0x197F) => 94,    (0x1980..0x19DF) => 95,
+        (0x1A00..0x1A1F) => 96,    (0x2C00..0x2C5F) => 97,
+        (0x2D30..0x2D7F) => 98,    (0x4DC0..0x4DFF) => 99,
+        (0xA800..0xA82F) => 100,   (0x10000..0x1007F) => 101,
+        (0x10080..0x100FF) => 101, (0x10100..0x1013F) => 101,
+        (0x10140..0x1018F) => 102, (0x10380..0x1039F) => 103,
+        (0x103A0..0x103DF) => 104, (0x10450..0x1047F) => 105,
+        (0x10480..0x104AF) => 106, (0x10800..0x1083F) => 107,
+        (0x10A00..0x10A5F) => 108, (0x1D300..0x1D35F) => 109,
+        (0x12000..0x123FF) => 110, (0x12400..0x1247F) => 110,
+        (0x1D360..0x1D37F) => 111, (0x1B80..0x1BBF) => 112,
+        (0x1C00..0x1C4F) => 113,   (0x1C50..0x1C7F) => 114,
+        (0xA880..0xA8DF) => 115,   (0xA900..0xA92F) => 116,
+        (0xA930..0xA95F) => 117,   (0xAA00..0xAA5F) => 118,
+        (0x10190..0x101CF) => 119, (0x101D0..0x101FF) => 120,
+        (0x102A0..0x102DF) => 121, (0x10280..0x1029F) => 121,
+        (0x10920..0x1093F) => 121, (0x1F030..0x1F09F) => 122,
+        (0x1F000..0x1F02F) => 122
+      }.freeze
+
+      UNICODE_MAX = 0xFFFF
+      UNICODE_RANGES = UNICODE_BLOCKS.keys.freeze
+
       def tag
         'OS/2'
       end
 
+      class << self
+        def encode(os2, subset)
+          ''.tap do |result|
+            result << [
+              os2.version, os2.ave_char_width, os2.weight_class,
+              os2.width_class, os2.type, os2.y_subscript_x_size,
+              os2.y_subscript_y_size, os2.y_subscript_x_offset,
+              os2.y_subscript_y_offset, os2.y_superscript_x_size,
+              os2.y_superscript_y_size, os2.y_superscript_x_offset,
+              os2.y_superscript_y_offset, os2.y_strikeout_size,
+              os2.y_strikeout_position, os2.family_class
+            ].pack('n*')
+
+            result << os2.panose
+
+            new_char_range = unicode_blocks_for(os2, os2.char_range, subset)
+            result << BinUtils
+                      .unpack_int(new_char_range.value, 32)
+                      .pack('N*').ljust(16, "\0")
```

Comment:
I think, it would be easier to supply `unpack_int` with a desired target size instead of manually padding the result. It also would make the method itself a bit simpler.